src/markov/mc_tools.jl

```diff
@@ -291,189 +291,155 @@ todense(::Type, A::Array) = A
-
-    simulate!(mc, X)
-    return X
-end
+### Returns
 
+- `X::Vector` : Vector containing the sample path, with length 
+ts_length
 """
-Simulate time series of state transitions of the Markov chain `mc`.
-
-##### Arguments
-
-- `mc::MarkovChain` : MarkovChain instance.
-- `ts_length::Int` : Length of each simulation.
-- `init::Int` : Initial state.
-- `;num_reps::Int(1)` : Number of repetitions of simulation.
-
-##### Returns
 
-- `X::Matrix{Int}` : Array containing the sample paths as columns, of shape
-(ts_length, k), where k = num_reps.
+function simulate(mc::MarkovChain, ts_length::Int;
+                  init::Int=rand(1:n_states(mc)))
+    X = Array(eltype(mc.state_values), ts_length, 1)
+    simulate!(mc, X; init=init)
+    return vec(X)
+end
 
-"""
-simulate(mc::MarkovChain, ts_length::Int, init::Int; num_reps::Int=1) =
-    simulate(mc, ts_length, [init], num_reps=num_reps)
 
 """
-Simulate time series of state transitions of the Markov chain `mc`.
+Fill `X` with sample paths of the Markov chain `mc` as columns.
+The resulting matrix has the state values of `mc` as elements.
 
-##### Arguments
+### Arguments
 
 - `mc::MarkovChain` : MarkovChain instance.
-- `ts_length::Int` : Length of each simulation.
-- `;num_reps::Union{Int, Void}(nothing)` : Number of repetitions of simulation.
-
-##### Returns
-
-- `X::Matrix{Int}` : Array containing the sample paths as columns, of shape
-(ts_length, k), where k = num_reps.
-
+- `X::Matrix` : Preallocated matrix to be filled with sample paths
+of the Markov chain `mc`. The element types in `X` should be the 
+same as the type of the state values of `mc`
+- `;init=rand(1:n_states(mc))` : Can be one of the following
+    - blank: random initial condition for each chain
+    - scalar: same initial condition for each chain
+    - vector: cycle through the elements, applying each as an
+      initial condition
 """
-function simulate(mc::MarkovChain, ts_length::Int; num_reps::Int=1)
-    init = rand(1:size(mc.p, 1), num_reps)
-    return simulate(mc, ts_length, init)
-end
 
-"""
-Fill `X` with sample paths of the Markov chain `mc` as columns.
-
-##### Arguments
+function simulate!(mc::MarkovChain, X::Matrix;
+                   init=rand(1:n_states(mc), size(X, 2)))
+    # Note: eltype(X) must be the same as eltype(mc.state_values)
+    if eltype(X) != eltype(mc.state_values)
+        throw(ArgumentError("Types in X must be the same as the types of the
+        Markov state values"))
+    end
 
-- `mc::MarkovChain` : MarkovChain instance.
-- `X::Matrix{Int}` : Preallocated matrix of integers to be filled with sample
-paths of the markov chain `mc`. The elements in `X[1, :]` will be used as the
-initial states.
+    ts_length, k = size(X)
+    
+    # if init is a vector, assign initial conditions to columns of X
+    # otherwise, just start each column at the same initial condition
+    real_init = Array(Int, k, )
+    if length(init) > 1
+        real_init = collect(take(cycle(init), k))
```

Comment:
Is there a reason that we use `collect(take(cycle(init), k))` here (as opposed to something simpler like filling the array)?

It seems like the reason to do this would be in case they gave us one number of initial conditions (2 for example) but wanted us to fill a matrix with a different number of columns (4 for example) -- In this case we would simulate 4 chains and start them with [first initial condition, second initial condition, first initial condition, second initial condition]. Is this what you had in mind? This should be elaborated on in the documentation, but I can imagine this surprising people.